packages/ui/src/components/TableV2/HeaderCell.tsx

```diff
@@ -0,0 +1,134 @@
+import styled from '@emotion/styled'
+import type { ReactNode } from 'react'
+import { Icon } from '../Icon'
+import { Stack } from '../Stack'
+
+const ArrowDownIcon = styled(Icon)``
+const ArrowUpIcon = styled(Icon)``
+
+const StyledIconContainer = styled(Stack)`
+  color: ${({ theme }) => theme.colors.neutral.textWeak};
+
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+  }
+
+  &[role*='button'] {
+    cursor: pointer;
+    user-select: none;
+  }
+
+  &[aria-sort] {
+    color: ${({ theme }) => theme.colors.primary.iconWeak};
+  }
+
+  &[aria-sort='ascending'] ${ArrowUpIcon} {
+    color: ${({ theme }) => theme.colors.primary.iconWeak};
+  }
+
+  &[aria-sort='descending'] ${ArrowDownIcon} {
+    color: ${({ theme }) => theme.colors.primary.iconWeak};
+  }
+`
+
+const SortIcon = () => (
+  <StyledIconContainer>
+    <ArrowUpIcon name="arrow-up" size={10} />
+    <ArrowDownIcon name="arrow-down" size={10} />
+  </StyledIconContainer>
+)
+
+type StyledHeaderCellProps = Pick<
+  HeaderCellProps,
+  'width' | 'maxWidth' | 'minWidth'
+>
+const StyledHeaderCell = styled('th', {
+  shouldForwardProp: prop => !['width', 'maxWidth', 'minWidth'].includes(prop),
+})<StyledHeaderCellProps>`
+  ${({ width, maxWidth, minWidth }) => `
+    ${width ? `width: ${width};` : ''}
+    ${maxWidth ? `max-width: ${maxWidth};` : ''}
+    ${minWidth ? `min-width: ${minWidth};` : ''}
+  `}
+
+  padding: ${({ theme }) => theme.space['1']};
+`
+
+const HeaderCellContent = styled('div')`
```

Comment:
Why not using a `Text` with prop `as="div"` so you don't need to add CSS related to text style